Mlem/Extensions/JSONDecoder+dateDecodingStrategyFormatters.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  JSONDecoder+dateDecodingStrategyFormatters.swift
+//  Mlem
+//
+//  Created by tht7 on 18/06/2023.
+//
+
+import Foundation
+
+extension DateFormatter {
+    static let expandedTWithTimezone: DateFormatter = {
+        var dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSSZ"
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        return dateFormatter
+    }()
+    
+    static let expandedT: DateFormatter = {
+        var dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSS"
+        dateFormatter.timeZone = TimeZone(identifier:"GMT")
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        return dateFormatter
+    }()
+    
+    static let standardT: DateFormatter = {
+        var dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss"
+        dateFormatter.timeZone = TimeZone(identifier:"GMT")
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        return dateFormatter
+    }()
+
+    static let standard: DateFormatter = {
+        var dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss"
+        dateFormatter.timeZone = TimeZone(identifier:"GMT")
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+        return dateFormatter
+    }()
+
+    static let yearMonthDay: DateFormatter = {
+        var dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
```

Comment:
which Lemmy instance have you seen returning this format from the API?

it'd be worth us having a list of any instances we come across who have these  _rogue_ responses so that we can ensure we test builds against them to make sure we're as compatible as possible

Reply:
I didn't keep a list but I'll start doing so